data/vibe/data/serialization.d

```diff
@@ -601,7 +645,15 @@ private template deserializeValueImpl(Serializer, alias Policy) {
 			static if (hasPolicyAttributeL!(ByNameAttribute, Policy, ATTRIBUTES)) {
 				return ser.deserializeValue!(string, ATTRIBUTES).to!T();
 			} else {
-				return cast(T)ser.deserializeValue!(OriginalType!T);
+				const auto value = cast(T)ser.deserializeValue!(OriginalType!T);
+
+				switch (value) {
+					static foreach (enumvalue; EnumMembers!T) {
+						case enumvalue: return value;
```

Comment:
This is going to have a problem with enums that have repeated members. This doesn't mean the value is invalid, just ambiguous.